src/bandersnatch/mirror.py

```diff
@@ -249,18 +251,73 @@ async def package_syncer(self, idx: int) -> None:
                 logger.debug(f"Package syncer {idx} emptied queue")
                 break
 
-            await package.sync()
+            try:
+                metadata = await package.fetch_metadata(attempts=3)
+                # Don't save anything if our metadata filters all fail.
```

Comment:
All this I want to eventually move back into a function on this side (as opposed to on the package)

Reply:
In the process of refactoring though ```Mirror``` might become a bit of a god-class.  Hopefully temporarily.